bin/hdfcoinc/pycbc_coinc_findtrigs

```diff
@@ -239,94 +252,153 @@ for tnum in template_ids:
     logging.info('Calculating Single Detector Statistic')
     s0g, s1g = rank_method.single(trigs0), rank_method.single(trigs1)
 
+    try:
+        s0gstat = s0g['snglstat'].copy()
+        s1gstat = s1g['snglstat'].copy()
+    except IndexError:
+        s0gstat = s0g.copy()
+        s1gstat = s1g.copy()
+
+    s0gsort = numpy.argsort(s0gstat)
+    s1gsort = numpy.argsort(s1gstat)
+
+    if not rank_method.single_increasing:
+        s0gsort = s0gsort[::-1]
+        s1gsort = s1gsort[::-1]
+
     # Loop over the single triggers and calculate the coincs they can
     # form
     start0 = 0
     while start0 < len(s0g):
         start1 = 0
+
+        end0 = start0 + args.batch_singles
+        if end0 > len(s0g):
+            end0 = len(s0g)
+
+        s0gid = s0gsort[start0:end0]
+
+        # Set the local parts of the single information we'll use
+        tid0 = tid0g[s0gid]
+        s0 = s0g[s0gid]
+        t0 = t0g[s0gid]
+
+        s1lims = {}
+        s1lower = {}
+        for kidx in range(1, len(threshes)):
+            # For each trigger in detector 0 get the limit in detector 1 to pass
+            # the current decimation threshold
+            s1lims[kidx] = rank_method.coinc_lim_for_thresh(s0, threshes[kidx])
+            if not rank_method.single_increasing:
+                s1lims[kidx] *= -1.
+            # subtract small amount to account for errors due to rounding
+            s1lims[kidx] -= 1e-6
+            # Get the minimum statistic required for all triggers
```

Comment:
maybe clarify to "...for all triggers at this decimation threshold"

Reply:
This comment has been updated